src/commands/shell.ts

```diff
@@ -0,0 +1,120 @@
+import { Command } from "../command";
+import { start } from "repl";
+import { yellow, red, bold } from "cli-color";
+import * as admin from "firebase-admin";
+import { getFirebaseConfig } from "../functionsConfig";
+import { getAccessToken } from "../api";
+import { getRefreshToken } from "../auth";
+import { requireAuth } from "../requireAuth";
+import { FirebaseError } from "../error";
+import { createContext, runInContext, Context, Script } from "vm";
+import { readFileSync, existsSync } from "fs";
+import fetch, { Response } from "node-fetch";
+import { DocumentSnapshot, Firestore, QuerySnapshot } from "@google-cloud/firestore";
+import api = require("../api");
+
+async function runScript(sandbox: Context, scriptPath: string) {
+  if (!existsSync(scriptPath)) {
+    throw new FirebaseError(`Script file ${bold(scriptPath)} does not exist.`);
+  }
+
+  try {
+    const script = new Script(readFileSync(scriptPath, { encoding: "utf8" }), {
+      filename: scriptPath,
+    });
+    return await Promise.resolve(script.runInContext(sandbox));
+  } catch (e) {
+    // show the full stacktrace for the user's custom script
+    throw new FirebaseError(`Error running script: ${e.stack}`);
+  }
+}
+
+function runRepl(sandbox: Context) {
```

Comment:
Can return types be added where necessary? `any` warnings can be ignored, but return types are useful.

Reply:
Done.